41-vaccine-temp-monitor/firmware/arduino/src/main.cpp

```diff
@@ -0,0 +1,528 @@
+    static const size_t numEnvVars = sizeof(envVars) / sizeof(envVars[0]);
+
+    // Allocate the environment variable manager.
+    if (!(envVarManager = NotecardEnvVarManager_alloc()))
+    {
+        notecard.logDebug("Failed to allocate env var manager.");
+    }
+
+    // Set the callback for the manager, and give it a pointer to our cache
+    // so that we can store the values of the environment variables we're
+    // fetching.
+    if (NEVM_SUCCESS != NotecardEnvVarManager_setEnvVarCb(envVarManager, envVarManagerCb, nullptr))
+    {
+        notecard.logDebug("Failed to set env var manager callback.");
+    }
+
+    if (NEVM_SUCCESS != NotecardEnvVarManager_fetch(envVarManager, envVars, numEnvVars))
+    {
+        notecard.logDebug("Failed to fetch environment variables.");
+    }
+}
+
+bool inHysteresis(void)
+{
+    bool result = true; // To mitigate I2C communication failure, we must assume
+                        // we are in hysteresis to sample Notecard again soon.
+
+    // Check status of OS pin
+    J *rsp = nullptr;
+    if (J *req = notecard.newRequest("card.aux"))
+    {
+        JAddStringToObject(req, "mode", "gpio");
+        if (J *usage = JAddArrayToObject(req, "usage"))
+        {
+            JAddItemToArray(usage, JCreateString("input-pullup")); // Aux 1
+            JAddItemToArray(usage, JCreateString(""));             // Aux 2
+            JAddItemToArray(usage, JCreateString(""));             // Aux 3
+            JAddItemToArray(usage, JCreateString(""));             // Aux 4
+            rsp = notecard.requestAndResponseWithRetry(req, RETRY_S);
+        }
+        else
+        {
+            notecard.deleteResponse(req);
+        }
+    }
+
+    // Parse response
+    if (rsp)
+    {
+        if (J *state = JGetObject(rsp, "state"))
+        {
+            if (J *aux1 = JGetArrayItem(state, 0))
+            {
+                result = JGetBool(aux1, "low");
+            }
+        }
+    }
+
+    // Clean up and exit
+    notecard.deleteResponse(rsp);
+    return result;
+}
+
+void setup()
+{
+    // Provide visual signal when the Host MCU is powered
+    ::pinMode(LED_BUILTIN, OUTPUT);
+    ::digitalWrite(LED_BUILTIN, HIGH);
+
+#ifndef NDEBUG
+#warning "Debug mode is enabled. Define NDEBUG to disable debug."
+    // Initialize Debug Output
+    usbSerial.begin(115200);
+    static const size_t MAX_SERIAL_WAIT_MS = 5000;
+    size_t begin_serial_wait_ms = ::millis();
+    while (!usbSerial && (MAX_SERIAL_WAIT_MS > (::millis() - begin_serial_wait_ms)))
+    {
+        ; // wait for debug serial port to connect. Needed for native USB
+    }
+    notecard.setDebugOutputStream(usbSerial);
+#endif
+
+    // Initialize LM75A
+    lm75a.begin();
+
+    // Initialize Notecard
+    notecard.begin();
+#ifdef ARDUINO_ARCH_ESP32
+    // Initialize Auxiliary Wi-Fi
+    aux_wifi.begin();
+#endif
+
+    // Configure Notecard
+    configureNotecard();
+
+    // Fetch Environment Variables
+    fetchEnvironmentVariables();

```

Comment:
So we only fetch environment variables at startup? Just checking that's the intent. No way to update the values without rebooting.

EDIT: Or does setup get run again every time we wake up? That may be worth clarifying with a comment, if so.

Reply:
Yes, due the nature of the application, it's a once and done. I will add a comment to this affect.

Reply:
After having seen it in context, I think the comment belongs in `README.md`.

Otherwise, this "problem" would plague nearly every aspect of this type of application.